src/autocomplete_test.js

```diff
@@ -196,6 +196,55 @@ module.exports = {
                 callback();
             }, 70);
         }
+    },
+    "test: slow and fast completers": function(done) {
+        var syncCompleter={
+            getCompletions: function(editor, session, pos, prefix, callback) {
+                callback(null,[{
+                    value: "test"
+                }]);
+            },
+            id: "asyncCompleter"
+        };
+
+        var asyncCompleter = {
+            getCompletions: function(editor, session, pos, prefix, callback) {
+                setTimeout(() => {
+                    callback(null,[{
+                        value: "some"
+                    }]);
+                }, 10);
+            },
+            id: "asyncCompleter"
+        };
+        var editor = initEditor("");
+        editor.completers=[
+            syncCompleter,asyncCompleter
+        ];
+        editor.resize(true);
+        editor.execCommand("insertstring", "o");
+        assert.notOk(!!editor.completer.popup);
+        setTimeout(function() {
+            assert.ok(editor.completer.popup.isOpen);
+            assert.equal(editor.completer.popup.renderer.scrollTop, 0);
+            editor.completer.popup.renderer.$loop._flush();
+            assert.equal(editor.completer.popup.renderer.scrollTop, 0);
+            assert.equal(editor.completer.popup.renderer.scroller.textContent, "some");
+            editor.onCommandKey(null, 0, 13);
```

Comment:
Can we either indicate in a comment which key corresponds with this keycode or use the `keys` map we have in the `lib` folder?

Reply:
changed all key emulation to use test/user.js, and fixed a bug in innerHTML emulation